### ISSUE TITLE
Quitting Shiny app in RStudio ends up at debugger prompt, when Shiny is installed with source refs

### ISSUE THREAD

**Issue by wch** [rstudio/rstudio#2410]
Closing a Shiny app drops to debugger prompt, when Shiny was installed with source refs. If Shiny was not installed with source refs, the app will exit normally (and not to a debugger prompt). This also does not happen when the Shiny app is run from R in a terminal.

### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.407
    OS Version      : macOS 10.13.3
    R Version       : 3.4.3

### Steps to reproduce the problem

```R
install.packages("shiny", type="source", INSTALL_opts="--with-keep.source")
shiny::runExample('01_hello')
# Close viewer window or press Stop button
```

It will drop you into a debugger:

![image](https://user-images.githubusercontent.com/86978/37478482-583562b0-2848-11e8-951e-fe11d68079e2.png)

*****

```
> sessioninfo::session_info()
─ Session info ────────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.4.3 (2017-11-30)
 os       macOS High Sierra 10.13.3   
 system   x86_64, darwin15.6.0        
 ui       RStudio                     
 language (EN)                        
 collate  en_US.UTF-8                 
 tz       America/Chicago             
 date     2018-03-15                  

─ Packages ────────────────────────────────────────────────────────────────────────────────
 package     * version    date       source                          
 clisymbols    1.2.0      2017-05-21 CRAN (R 3.4.0)                  
 digest        0.6.15     2018-01-28 cran (@0.6.15)                  
 htmltools     0.3.6      2017-04-28 CRAN (R 3.4.0)                  
 httpuv        1.3.6.2    2018-03-02 CRAN (R 3.4.3)                  
 jsonlite      1.5        2017-06-01 CRAN (R 3.4.0)                  
 mime          0.5        2016-07-07 CRAN (R 3.4.0)                  
 R6            2.2.2      2017-06-17 CRAN (R 3.4.0)                  
 Rcpp          0.12.16    2018-03-13 cran (@0.12.16)                 
 rsconnect     0.8.5      2017-08-23 CRAN (R 3.4.1)                  
 sessioninfo   1.0.0      2017-06-21 CRAN (R 3.4.1)                  
 shiny       * 1.0.5      2017-08-23 CRAN (R 3.4.3)                  
 withr         2.1.1.9000 2018-02-27 Github (jimhester/withr@5d05571)
 xtable        1.8-2      2016-02-05 CRAN (R 3.4.0)                  
 yaml          2.1.17     2018-02-27 CRAN (R 3.4.3)                  
```

**Comment by wch** [comment]
@bborgesr said she was unable to reproduce this, so it's possible it's something odd with my local configuration.

**Comment by ronblum** [comment]
@wch I was able to reproduce it by setting `Debug -> On Error -> Break In Code`

**Comment by wch** [comment]
@ronblum Does it reproduce the behavior where, when shiny is installed with source refs, it goes to the debugger prompt, but when shiny is installed without source refs, it returns to the regular R prompt?

FWIW, I currently have `Debug -> On Error -> Message Only`

**Comment by wch** [comment]
Update: I switched from Message Only to Break In Code, and it continued to go to the debugger prompt when I closed the app. However, when I switched *back* to Message Only, it stopped going to the debugger prompt, and returned to the regular R prompt.

I think this may mean that, previously, my IDE was *displaying* Message Only, but it was really in the Break In Code state.

When I have Break In Code selected, I still find that when Shiny is installed with source refs, it goes to the debugger when I quit the app, and when it is installed without source refs, it goes to the regular R prompt. Is that the expected behavior?
